### ISSUE TITLE
1.1.310  - Possible regression with a string format

### ISSUE THREAD

**Issue by sylvainmouquet** [microsoft/pyright#5167]
**Describe the bug**
```
 commons/monitoring/datadog.py:30:13 - error: String literal is unterminated
 commons/monitoring/datadog.py:28:16 - error: "(" was not closed
 commons/monitoring/datadog.py:29:62 - warning: Unsupported escape sequence in string literal (reportInvalidStringEscapeSequence)
```

**To Reproduce**
```python
    logger.info(
        f"\nENABLE_DATADOG_TRACING: {ENABLE_DATADOG_TRACING} \
            \nENVIRONMENT_NAME: {ENVIRONMENT_NAME} \
            \nPROJECT_NAME: {PROJECT_NAME} \
            \nVERSION: {VERSION} \
        "
    )
```

**Expected behavior**
The code is valid

**Comment by erictraut** [comment]
Thanks for reporting. Here's a minimal, self-contained code sample that exhibits the problem:

```python
f"x: {0} \
    y: {0}
"
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.311, which I just published. It will also be included in a future release of pylance.

**Comment by longwuyuan** [comment]
Hi,

I am experiencing this problem even on the latest pyright.

- I have pyright installed for neovim via lazy-vim. I enabled LazyVim-Extras using lang-python  https://www.lazyvim.org/extras/lang/python

 ![Image](https://github.com/user-attachments/assets/65a133e6-4808-44c4-bb93-1e95f19c3448)

- The error looks like this

![Image](https://github.com/user-attachments/assets/650b6922-e37d-4592-b8d8-5cbe0ff1ceed)


- The python code
```
import re

with open("myfile.txt", mode="w+") as f:
    f.write("Hello World!\n")
    f.write("Welcome to Python!\n")
    f.write(
        "This is a C class ipaddress 192.168.193.1. And this 172.16.172.1 is a B class ipaddress\n"
    )
    f.seek(0)
    for line in f.readlines():
        print("line from readlines:", line)
        result = re.findall("\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}", line)
        print(result)


```

**Comment by longwuyuan** [comment]
cc @erictraut

**Comment by erictraut** [comment]
@longwuyuan, this issue was fixed many months ago. If you think that you've identified a bug, please open a new issue.

Looking at your sample above, this doesn't appear to be a bug. You're attempting to use escape sequences that are not supported by Python. For a list of escape sequences supported by Python, refer to [this documentation](https://docs.python.org/3/reference/lexical_analysis.html#escape-sequences). Note also that this official documentation says:

> Unrecognized escape sequences produce a [SyntaxWarning](https://docs.python.org/3/library/exceptions.html#SyntaxWarning). In a future Python version they will be eventually a [SyntaxError](https://docs.python.org/3/library/exceptions.html#SyntaxError).

If your intent is to generate a backslash, you should use a double backslash in your string literal — or switch to a raw string.

**Comment by longwuyuan** [comment]
@erictraut thank you very much. I understand now.
